test/https.ts

```diff
@@ -1,37 +1,40 @@
 import test from 'ava';
 import got, {CancelableRequest} from '../source';
-import withServer from './helpers/with-server';
+import {withHttpsServer} from './helpers/with-server';
 import {DetailedPeerCertificate} from 'tls';
 import pEvent from 'p-event';
+import pify = require('pify');
+import pem = require('pem');
 
-test('https request without ca', withServer, async (t, server, got) => {
+const createPrivateKey = pify(pem.createPrivateKey);
+const createCSR = pify(pem.createCSR);
+const createCertificate = pify(pem.createCertificate);
+const createPkcs12 = pify(pem.createPkcs12);
+
+test('https request without ca', withHttpsServer(), async (t, server, got) => {
 	server.get('/', (_request, response) => {
 		response.end('ok');
 	});
 
-	t.truthy((await got.secure({
+	t.truthy((await got({
 		https: {
+			certificateAuthority: [],
 			rejectUnauthorized: false
 		}
 	})).body);
 });
 
-test('https request with ca', withServer, async (t, server, got) => {
+test('https request with ca', withHttpsServer(), async (t, server, got) => {
```

Comment:
Can you define `withHttpsServer` so `()` is not necessary to call every time?

Reply:
I could, but there are some tests that give parameters to it, so I thought that leaving it always as a function call was going to be cleaner.

Reply:
Cool. I'm fine with that :)